src/app/modules/photo-uploader/components/uploader-content/uploader-content.component.ts

```diff
@@ -0,0 +1,190 @@
+import { MatDialog } from '@angular/material';
+import { UploadPhotoFileItemService } from '../../services/upload-photo-file-item.service';
+import { UploadFileService } from '../../services/upload-file.service';
+
+@Component({
+  selector: 'app-uploader-content',
+  templateUrl: './uploader-content.component.html',
+  styleUrls: ['./uploader-content.component.scss']
+})
+export class UploaderContentComponent implements OnInit, OnDestroy {
+
+  public uploader: FileUploader;
+  baseUrl = environment.apiUrl;
+  uploadPhotoForm: FormGroup;
+  photoUploaderModels: PhotoUploaderModel[];
+  public photoUploaderRemoveAllPhotosSubscription: any;
+  public photoUploaderRemoveChosenPhotoSubscription: any;
+  public photoUploaderPropagateNewPhotosSubscription: any;
+
+  public hasBaseDropZoneOver = false;
+  public photoHasUploaded = false;
+
+  constructor(
+    private photoEventService: PhotoEventService,
+    private localStorageService: UploadPhotoLocalStorageService,
+    private fileItemService: UploadPhotoFileItemService,
+    private dialog: MatDialog,
+    private uploadFileService: UploadFileService) { }
+
+  ngOnInit() {
+    this.photoUploaderModels = new Array<PhotoUploaderModel>();
+    this.initUploader();
+    this.subscribeEvents();
+  }
+
+  ngOnDestroy(): void {
+    this.photoUploaderRemoveAllPhotosSubscription.unsubscribe();
+    this.photoUploaderRemoveChosenPhotoSubscription.unsubscribe();
+    this.photoUploaderPropagateNewPhotosSubscription.unsubscribe();
+  }
+
+  initUploader() {
+    this.uploader = new FileUploader({
+      isHTML5: true,
+      allowedFileType: ['image'],
+      disableMultipart: true,
+      autoUpload: false,
+      removeAfterUpload: false
+    });
+  }
+
+  onChangePreviewImages() {
+    this.propagateNewPhotosProcessing();
+  }
+
+  GetPhotoUploaderModel(fileItem: FileItem): PhotoUploaderModel {
+    return this.fileItemService.GetPhotoUploaderModel(fileItem, this.photoUploaderModels);
+  }
+
+  fileOverBase(e: any): void {
+    this.hasBaseDropZoneOver = e;
+  }
+
+  private updateUploaderQueue(files: File[]) {
+    if (files.length > 0) {
+      this.uploader.addToQueue(files);
+      this.propagateNewPhotosProcessing();
+    }
+  }
+
+  private propagateNewPhotosProcessing() {
+    this.photoHasUploaded = true;
+    this.photoEventService.emitPhotoUploaded(true);
+    this.photoEventService.emitPhotoUploaderCountOfAcctualPhotos(this.uploader.queue.length);
+    this.fileItemService.prepareIndexForPhotoUploader(this.uploader);
+    this.preparePhotoUploaderModel();
+    this.saveNewPhoto();
+  }
+
+  private saveNewPhoto() {
+    const formDataToPost = this.prepareNewPhotoDataToSave();
+    this.uploadFileService.uploadFile(formDataToPost).subscribe(() => {
+      // to do after save new picture
+    }, error => {
+      console.log(error);
+    });
+  }
+
+  private prepareNewPhotoDataToSave() {
+    const formDataToPost = new FormData();
+    formDataToPost.append('index', '1');
```

Comment:
What about the rest properties from PhotoUploaderModel such as: photoTitle, photoDescription and photoTags?

Reply:
its will be extended in the features with sc path for fileUpload. now it's only skeleton :)